gnocchi/incoming/s3.py

```diff
@@ -175,3 +170,28 @@ def process_measure_for_metrics(self, metric_ids):
 
         # Now clean objects
         s3.bulk_delete(self.s3, self._bucket_name_measures, all_files)
+
+    @contextlib.contextmanager
+    def process_measures_for_sack(self, sack):
+        measures = defaultdict(self._make_measures_array)
+        files = self._list_measure_files((str(sack),))
+        for f in files:
+            try:
+                sack, metric_id, measure_id = f.split("/")
+                metric_id = uuid.UUID(metric_id)
+            except ValueError:
+                LOG.warning("Unable to parse measure file name %s", f)
+                continue
+
+            response = self.s3.get_object(
+                Bucket=self._bucket_name_measures,
+                Key=f)
+            measures[metric_id] = numpy.concatenate((
+                measures[metric_id],
+                self._unserialize_measures(f, response['Body'].read())
+            ))
+
+        yield measures
+
+        # Now clean objects
+        s3.bulk_delete(self.s3, self._bucket_name_measures, files)
```

Comment:
files may be a superset of actual processed measures... (assuming it fails to convert to UUID)

Reply:
Is this a problem? I imagine it's better to clean the wrong file than converting a bad file over and over again.

Reply:
i agree. we should do the same for ceph then, as it only deletes what it processes.

Reply:
same for file?

Reply:
I'll do it for Ceph. For files, it's not really possible as it's a directory with wrong format not matching a metric, so it's different.